server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+	var schemas []models.Schema
+	cursor, err := schemasCollection.Aggregate(context.TODO(), mongo.Pipeline{
+		bson.D{{"$match", bson.D{{"$or", []interface{}{
+			bson.D{{"is_deleted", false}},
+			bson.D{{"is_deleted", bson.D{{"$exists", false}}}},
+		}}}}},
+	})
+
+	if err != nil {
+		return []models.ExtendedSchema{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemas); err != nil {
+		return []models.ExtendedSchema{}, err
+	}
+
+	var schemaVersion models.SchemaVersion
+	var extednedSchemas []models.ExtendedSchema
+	for _, schema := range schemas {
+		for _, version := range schema.Versions {
+			filter := bson.M{"_id": version}
+			err = schemasVersionCollection.FindOne(context.TODO(), filter).Decode(&schemaVersion)
+			if err != nil {
+				return []models.ExtendedSchema{}, err
+			}
+			if schemaVersion.VersionNumber == 1 {
+				extSchema := models.ExtendedSchema{
+					ID:            schema.ID,
+					Name:          schema.Name,
+					Type:          schema.Type,
+					CreatedByUser: schemaVersion.CreatedByUser,
+					CreationDate:  schemaVersion.CreationDate,
+				}
+				extednedSchemas = append(extednedSchemas, extSchema)
+			}
+		}
+
+	}
+
+	if len(extednedSchemas) == 0 {
+		return []models.ExtendedSchema{}, nil
+	} else {
+		return extednedSchemas, nil
+	}
+}
+
+func (sh SchemasHandler) GetAllSchemaVersionsByVersionNumber(versionNumber int) ([]models.SchemaVersion, error) {
+	var schemasVersion []models.SchemaVersion
+
+	filter := bson.M{"version_number": versionNumber}
+	cursor, err := schemasVersionCollection.Find(context.TODO(), filter)
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemasVersion); err != nil {
+		return schemasVersion, err
+	}
+
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+
+	if len(schemasVersion) == 0 {
+		return []models.SchemaVersion{}, nil
+	} else {
+		return schemasVersion, nil
+	}
+}
+
+func (sh SchemasHandler) FindAndDeleteSchema(schemaName string) error {
+	var schema models.Schema
+	filter := bson.M{"name": schemaName}
+	err := schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil {
+		return err
+	}
+	schemaVersions := schema.Versions
+	var schemaVersion models.SchemaVersion
+	for _, version := range schemaVersions {
+		filterVersion := bson.M{"_id": version}
+		err = schemasVersionCollection.FindOneAndDelete(context.TODO(), filterVersion).Decode(&schemaVersion)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = schemasCollection.DeleteOne(context.TODO(), filter)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
+	var body models.CreateNewSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.Name)
+	err := validateSchemaName(schemaName)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
```

Comment:
this is not 500 error this is an error we want the user will see in order to understand his mistake.
So status code should be USER_SHOWABLE_STATUS from the config